libp2p/switch.nim

```diff
@@ -62,10 +62,15 @@ proc secure(s: Switch, conn: Connection): Future[Connection] {.async, gcsafe.} =
     raise newException(CatchableError, "No secure managers registered!")
 
   let manager = await s.ms.select(conn, s.secureManagers.mapIt(it.codec))
-  if manager.len == 0:
+  if manager.len == 0 or manager == "na":
```

Comment:
We shouldn't ever get "na" here, if you are something is wrong, but "na" is internal to multistream and shouldn't be returned.

Reply:
We do get to "na" indeed tho, I didn't add it just for the sake of it.

Reply:
If you think that the remotes are doing it wrong, we could simply ignore na but still check that `filterIt`, which btw should just be `find`, finds the right codec, if not we throw.

Reply:
here is why `na` will be returned... because of one of the things I dislike of nim, the `result` construct.
https://github.com/status-im/nim-libp2p/blob/master/libp2p/multistream.nim#L73-L86

if the first condition fails and there is only 1 proto.. the rest of the code won't run (the `for`) and leave `na` as result...

Reply:
Yeah, I think this has been there forever, it doesn't look like you added it, should be changed tho, doesn't make sense to return it here.

Reply:
Ok I will address it in this PR, if that is ok ?